src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforceStreamingSource.java

```diff
@@ -112,4 +129,18 @@ public Schema outputSchema(SalesforceStreamingSourceConfig config) throws Except
     return SalesforceSchemaUtil.getSchema(authenticatorCredentials,
                                           SObjectDescriptor.fromQuery(query));
   }
+
+  private void recordLineage(StreamingSourceContext context, String outputName, Schema tableSchema,
+                             String operationName, String description)
+    throws DatasetManagementException, TransactionFailureException {
+    Preconditions.checkNotNull(tableSchema, "schema for output %s is null.", outputName);
+    Preconditions.checkNotNull(tableSchema.getFields(), "schema.getFields() for output %s is null.", outputName);
```

Comment:
Same here, don't use Precondition, if will fail the pipeline, use if and log a message if schema is null